source/_localCaptioner/__init__.py

```diff
@@ -0,0 +1,197 @@
+	mem = wx.MemoryDC(bmp)
+
+	# Copy the specified screen region to the memory bitmap
+	mem.Blit(0, 0, width, height, wx.ScreenDC(), x, y)
+
+	# Convert the bitmap to an image object for more flexible operations
+	image = bmp.ConvertToImage()
+
+	# Create a byte stream object to save image data as binary data
+	body = io.BytesIO()
+
+	# Save the image to the byte stream in JPEG format
+	image.SaveFile(body, wx.BITMAP_TYPE_JPEG)
+
+	# Read the binary image data from the byte stream
+	imageData = body.getvalue()
+	return imageData
+
+
+def _caption(captioner: ImageCaptioner, imageData: bytes) -> None:
+	"""Generate a caption for the given image data.
+
+	Args:
+		captioner: The captioner instance to use for generation.
+		imageData: The image data to caption.
+	"""
+	try:
+		description = captioner.generateCaption(image=imageData)
+		ui.message(description)
+		api.copyToClip(text=description, notify=False)
+	except Exception as e:
+		# Translators: error message when fail to generate caption
+		ui.message(_("fail to generate caption"))
+		log.error(e)
+
+
+class LocalCaptioner:
+	"""module for local image caption functionality.
+
+	This module provides image captioning using local ONNX models.
+	It can capture screen regions and generate descriptive captions.
+	"""
+
+	def __init__(self) -> None:
+		self.isModelLoaded = False
+		self.captioner: ImageCaptioner | None = None
+
+		loadModelWhenInit = config.conf["captionLocal"]["loadModelWhenInit"]
+		# Load model when initializing (may cause high memory usage)
+		if loadModelWhenInit:
+			threading.Thread(target=self._loadModel, daemon=True).start()
+
+	def terminate(self):
+		self.captioner = None
+
+	def runCaption(self, gesture) -> None:
+		"""Script to run image captioning on the current navigator object.
+
+		Args:
+			gesture: The input gesture that triggered this script.
+		"""
+		imageData = _screenshotNavigator()
+
+		if not self.isModelLoaded:
+			self._loadModel()
+
+		imageThread = threading.Thread(target=_caption, args=(self.captioner, imageData))
+		# Translators: Message when starting image recognition
+		ui.message(_("getting Image description..."))
+		imageThread.start()
+
+	def _loadModel(self) -> None:
+		"""Load the ONNX model for image captioning.
+
+		Raises:
+			Exception: If the model cannot be loaded.
+		"""
+		# Translators: Message when loading the model
+		ui.message(_("loading model..."))
+
+		try:
+			localModelDirPath = config.conf["captionLocal"]["localModelPath"]
+			encoderPath = f"{localModelDirPath}/onnx/encoder_model_quantized.onnx"
+			decoderPath = f"{localModelDirPath}/onnx/decoder_model_merged_quantized.onnx"
+			configPath = f"{localModelDirPath}/config.json"
+
+			self.captioner = ImageCaptioner(
+				encoder_path=encoderPath,
+				decoder_path=decoderPath,
+				config_path=configPath,
+			)
+			self.isModelLoaded = True
+			# Translators: Message when successfully load the model
+			ui.message(_("image captioning on"))
```

Comment:
move this to an else block after the exception as a try-except-else, as this shouldn't raise an exception